File: mobile/lib/models/product.dart
class Product {
  final String name;
  final int id;
  final double price;
  final String origin;
  final String image_url;

  const Product({
    this.name,
    this.id,
    this.price,
    this.origin,
    this.image_url,
  });

  static Product fromJson(json) {
    return Product(
      name: json['name'],
      id: json['id'],
      price: json['price'],
      origin: json['origin'],
      image_url: json['image_url'],
    );
  }
}

File: market/dto/product.go
package dto

type NewProductRequest struct {
	Name     string  `json:"name"  binding:"required,min=3,max=70"`
	Price    float64 `json:"price" binding:"required,min=0.01"`
	Origin   string  `json:"origin" binding:"required,min=2,max=50"`
	ImageUrl string  `json:"image_url" binding:"required,min=10,max=250"`
}

type UpdateProductRequest struct {
	ID int64 `json:"id" binding:"required,min=1"`
	NewProductRequest
}
